src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -47,52 +47,102 @@ For the complete working example, including required imports, obtaining the file
 
 ## Create a record with an associated file
 
-First upload the file to Storage, then run the following mutation with the GraphQL API to create a record and associate the file with the record.
+First create a record via the GraphQL API, then upload the file to Storage, and finally add the association between the record and file. Use the following example with the GraphQL API and Storage categories to create a record and associate the file with the record.
 
 <Callout>
 
-Note: In the case of adding a file to an existing record (such as when a user uploads a profile picture to an existing profile), first create the record and exclude the optional file reference, then follow the steps under [Add or update a file for an associated record](#add-or-update-a-file-for-an-associated-record) below.
+The API record's `id` is prepended to the Storage file name to ensure uniqueness. If this is excluded, multiple API records could then be associated with the same file key unintentionally.
 
 </Callout>
 
 <BlockSwitcher>
 <Block name="TypeScript">
 
 ```ts
-const result = await Storage.put(file.name, file, {
-  contentType: "image/png", // contentType is optional
+const createSongDetails: CreateSongInput = {
+  name: `My first song`,
+};
+
+// Create the API record:
+const response = await API.graphql<GraphQLQuery<CreateSongMutation>>({
+  query: mutations.createSong,
+  variables: { input: createSongDetails },
 });
 
+const _song = response?.data?.createSong;
```

Comment:
Why is this variable prepended with an underscore? Is there something special about it vs all the others?

Reply:
I'd previously been using this naming convention, and later switched. Will make this consistent with the other variables.

Reply:
This is not needed, will update.